File: web/src/main/java/org/mitre/iwp/web/model/ServiceResponse.java
/* 
* NOTICE
*  
* This software (or technical data) was produced for the U. S. Government
* and is subject to the Rights in Data-General Clause 52.227-14, Alt. IV
* (May 2014) – Alternative IV (Dec 2007)
* 
* (c) 2024 The MITRE Corporation. All Rights Reserved.
*/ 

package org.mitre.iwp.web.model;

import com.google.gson.annotations.Expose;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;

@MappedSuperclass
public class ServiceResponse {
    @Id
    @GeneratedValue
    private Long imageId;
    public Long getImageId() { return this.imageId; }
    public void setImageId(Long imageId) { this.imageId = imageId; }

    public void setError(String error) {
        this.error = error;
    }
    public String getError() {
        return error;
    }

    public void setStatus(ServiceCallStatus status) { this.status = status; }
    public ServiceCallStatus getStatus() { return this.status; }

    public void setEbtsImageId(Long ebtsImageId) { this.ebtsImageId = ebtsImageId; }
    public Long getEbtsImageId() { return this.ebtsImageId; }

    @Expose
    String error;
    @Expose
    String imagePath;
    @Expose
    ServiceCallStatus status;
    @Expose
    Long ebtsImageId;
}

File: web/client/src/app/types/service-response.ts
/* 
* NOTICE
*  
* This software (or technical data) was produced for the U. S. Government
* and is subject to the Rights in Data-General Clause 52.227-14, Alt. IV
* (May 2014) – Alternative IV (Dec 2007)
* 
* (c) 2024 The MITRE Corporation. All Rights Reserved.
*/ 

export enum ServiceStatus {
  NotSent = 'NotSent',
  Pending = 'Pending',
  Error = 'Error',
  Received = 'Received'
}

export class ServiceResponse {
  // imageId is the modified id number
  // imageId and ebtsImageId can be the same number it's not a resubmitted image.
  imageId = '';
  error = '';
  status: ServiceStatus;
  ebtsImageId = '';
  isResubmit = false;

  constructor(
    imageId: string,
    error: string,
    status: ServiceStatus,
    ebtsImageId: string,
    isResubmit: boolean) {
    this.imageId = imageId;
    this.error = error;
    this.status = status;
    this.ebtsImageId = ebtsImageId;
    this.isResubmit = isResubmit;
  }
}
